lumen/ai/agents.py

```diff
@@ -581,6 +582,8 @@ async def _create_valid_sql(
             pipeline = await get_pipeline(
                 source=sql_expr_source, table=expr_slug, sql_transforms=sql_transforms
             )
+            if len(pipeline.data) == 0:
+                raise ValueError("No data was returned from the SQL query.")
```

Comment:
This seems like a condition that can actually happen and be a perfectly valid result. I'll remove and then you can try to convince me why this is needed later.

```suggestion
```